integreat_cms/api/v3/socialmedia_headers.py

```diff
@@ -0,0 +1,298 @@
+"""
+This module includes functions related to the social media headers API endpoint.
+"""
+from __future__ import annotations
+
+import logging
+from html import unescape
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+from django.shortcuts import render
+from django.utils.html import strip_tags
+from django.utils.translation import activate
+
+from integreat_cms.cms.models.events.event_translation import EventTranslation
+from integreat_cms.cms.models.languages.language import Language
+from integreat_cms.cms.models.pois.poi_translation import POITranslation
+from integreat_cms.cms.models.push_notifications.push_notification_translation import (
+    PushNotificationTranslation,
+)
+from integreat_cms.cms.models.regions.region import Region
+
+from ...cms.models import PageTranslation
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from django.http import HttpRequest, HttpResponse
+
+logger = logging.getLogger(__name__)
+
+
+def socialmedia_headers(
+    request: HttpRequest, path: Optional[str] = None
+) -> HttpResponse:
+    """
+    Detects the path context and renders the socialmedia headers.
+    Available contexts are:
+    - root e.g. /, /landing, /disclaimer, /landing/de
+    - region root e.g. /testumgebung, /testumgebung/de, /testumgebung/de/events
+    - page e.g. /testumgebung/de/willkommen, /testumgebung/de/events/event
+
+    :param request: The request that has been sent to the Django server
+    :param path: The path of the page the social meddia headers should be created for
+
+    :return: HTML meta headers required by social media platforms
+    """
+    slugs = list(filter(bool, path.split("/"))) if path else []
+
+    potential_region_slug = slugs[0] if len(slugs) > 0 else None
+    region = (
+        Region.objects.filter(slug=potential_region_slug).first()
+        if potential_region_slug
+        else None
+    )
+
+    potential_language = (
+        Language.objects.filter(slug=slugs[1]).first() if len(slugs) > 1 else None
+    )
+    language = potential_language or Language.objects.get(slug="de")
```

Comment:
We have "invalid" urls such as
- https://webnext.integreat.app/testumgebung/test
- https://webnext.integreat.app/testumg/de

And what you mean by throw an 404 is, that we do not want that these pages are shown in the search results and the server does not return any headers, right?

Reply:
Initially, I thought maybe it makes sense to show something like "Region not found"/"Language not available" as title and maybe a special error image for the social media preview to indicate that the links are (no longer) valid, but if this means that an existing social media preview can cause a page to be indexed by search engines, then it's not that optimal and it would be better to just not show any social media headers in this case.

Reply:
I've thought about this again. This would cause problems with urls where we detect the language and then redirect. Consider https://webnext.integreat.app/testumgebung or just https://webnext.integreat.app/ (which is the most crucial root url). These are both definitely valid. I see the comfort of having a region not found or language not found preview. But I'm not sure if this can actually be implemented.